File: frontend/src/types/part.ts

export interface GeometryNode {
  z: number;
  x: number;
}

export interface GeometrySegment {
  sequence: number;
  type: 'line' | 'arc_cw' | 'arc_ccw';
  start: GeometryNode;
  end: GeometryNode;
  text: string | null;
}

export interface DigitalRepresentation {
  overall_length: number;
  max_diameter: number;
  geometry: GeometrySegment[];
  ocr_raw?: any[];
}

export interface Part {
  id: string;
  drawing_id: string;
  part_name: string;
  digital_representation: DigitalRepresentation;
  created_at: string;
}

export interface Feature {
  id: string;
  part_id: string;
  feature_type: 'step' | 'groove' | 'thread' | 'chamfer' | 'radius' | 'hole';
  start_z: number;
  start_x: number;
  end_z: number;
  end_x: number;
  parameters?: Record<string, any>;
}

export interface MachiningOperation {
  sequence: number;
  operation: 'facing' | 'rough_turning' | 'finish_turning' | 'grooving' | 'threading' | 'inspection';
  description: string;
  tool_code: string;
  tool_name: string;
  cutting_speed: number;
  feed_rate: number;
  depth_of_cut: number;
  rpm: number;
  coolant: boolean;
  estimated_time: number;
}

export interface VerificationCheck {
  name: string;
  passed: boolean;
}

export interface VerificationResults {
  checks: VerificationCheck[];
  warnings: string[];
  errors: string[];
  confidence_score: number;
  explanation: string;
}

export interface ManufacturingPlan {
  id: string;
  part_id: string;
  material: string;
  machine_tool: string;
  operations: MachiningOperation[];
  gcode_program: string | null;
  verification_results: VerificationResults | null;
  created_at: string;
}

export interface AnalysisResult {
  drawing_id: string;
  status: 'completed' | 'failed';
  part: Part;
  features: Feature[];
  plan: ManufacturingPlan;
}

File: backend/app/api/schemas/part.py
from datetime import datetime
from typing import List, Dict, Any, Optional
from pydantic import BaseModel, ConfigDict

class FeatureResponse(BaseModel):
    id: str
    part_id: str
    feature_type: str
    start_z: float
    start_x: float
    end_z: float
    end_x: float
    parameters: Optional[Dict[str, Any]] = None

    model_config = ConfigDict(from_attributes=True)

class PlanResponse(BaseModel):
    id: str
    part_id: str
    material: str
    machine_tool: str
    operations: List[Dict[str, Any]]
    gcode_program: Optional[str] = None
    verification_results: Optional[Dict[str, Any]] = None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)

class PartResponse(BaseModel):
    id: str
    drawing_id: str
    part_name: str
    digital_representation: Dict[str, Any]
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)

class AnalysisResponse(BaseModel):
    drawing_id: str
    status: str  # "completed", "failed"
    part: PartResponse
    features: List[FeatureResponse]
    plan: PlanResponse
